src/contracts/core/DelegationManager.sol

```diff
@@ -510,199 +505,178 @@ contract DelegationManager is
         delegatedTo[staker] = operator;
         emit StakerDelegated(staker, operator);
 
+        // read staker's delegatable shares and strategies to add to operator's delegatedShares
+        // and also update the staker depositScalingFactor for each strategy
         (IStrategy[] memory strategies, uint256[] memory shares) = getDelegatableShares(staker);
+        uint64[] memory totalMagnitudes = allocationManager.getTotalMagnitudes(operator, strategies);
 
-        for (uint256 i = 0; i < strategies.length;) {
+        for (uint256 i = 0; i < strategies.length; ++i) {
             // forgefmt: disable-next-item
-            _increaseOperatorShares({
+            _increaseDelegation({
                 operator: operator, 
                 staker: staker, 
-                strategy: strategies[i], 
-                shares: shares[i]
+                strategy: strategies[i],
+                existingPrincipalShares: 0,
+                addedShares: shares[i],
+                totalMagnitude: totalMagnitudes[i]
             });
-
-            unchecked {
-                ++i;
-            }
         }
     }
 
     /**
-     * @dev commented-out param (middlewareTimesIndex) is the index in the operator that the staker who triggered the withdrawal was delegated to's middleware times array
-     * This param is intended to be passed on to the Slasher contract, but is unused in the M2 release of these contracts, and is thus commented-out.
+     * @dev This function completes a queued withdrawal for a staker.
+     * This will apply any slashing that has occurred since the the withdrawal was queued. By multiplying the withdrawl's
+     * delegatedShares by the operator's total magnitude for each strategy
```

Comment:
typo: `withdrawl` => `withdrawal`
also seems weird that there is a period instead of a comma here?